devel/query.lua

```diff
@@ -325,6 +329,23 @@ function runOnce(caller)
     return true
 end
 
+function toType(str)
+    if str ~= nil then
+        if str == "true" then
+            return true
+        elseif str == "false" then
+            return false
+        elseif tonumber(str) then
+            return tonumber(str)
+        elseif string.find(str, "nil") then
```

Comment:
Is there a reason you want to accept any string containing "nil" here? i.e. why not `elseif str == "nil"` like the cases for true/false above?

Part of me would also like to find a way to allow searching for strings that happen to also be valid booleans/numbers (e.g. the string "true"). I'm not sure if that's really important, but one way could be to allow an escape character (like `\`) at the start of `str`, and remove a single occurrence of it prior to calling `tostring()` below.

(Incidentally, the `tostring` call is redundant if `str` is already a string.)

Reply:
Oh yea, wups. I didn't even think about it.

Not sure I understand your idea. But if it puts the code for conversion inside the utils codebase then I think it's worth exploring; presumably allowing for args={*} to already be filled with correctly typed table entries.

About the str specifically I threw it through tostring just in case because I mostly just wasn't comfortable assuming it was in fact a string I like to think less and just be careful with dynamically typed languages.